fs_mark/fs_mark.py

```diff
@@ -0,0 +1,102 @@
+import trunner
+
+from trunner.ctx import TestContext
+from trunner.dut import Dut
+from trunner.types import TestResult, Result, Status
+
+
+def harness(dut: Dut, ctx: TestContext):
+    results = []
+    test_name = []
+
+    NAME = r"(.+?)(?P<name>/usr/bin/fs_mark.+?)\r+\n"
+    MSG_LINE = r"(\s+)(?P<line>([\d\.]+\s+){23})\r+\n"
+    FINAL = r"(?P<final>Insufficient free space.+?)\r+\n"
+    END = r"(?P<end>Average Files/sec:.+?)\r+\n"
+
+    f_use = None
+    count = None
+    size = None
+    files_sec = None
+    app_overhead = None
+    test_status = None
+    result = None
+    target = trunner.ctx.target.name
+
+    timing_dict = {
+            ('ia32-generic-qemu', 'creatMin'): [0, 90000],
+            ('ia32-generic-qemu', 'creatAvg'): [0, 90000],
+            ('ia32-generic-qemu', 'creatMax'): [0, 90000],
+            ('ia32-generic-qemu', 'writeMin'): [0, 90000],
+            ('ia32-generic-qemu', 'writeAvg'): [0, 90000],
+            ('ia32-generic-qemu', 'writeMax'): [0, 90000],
+            ('ia32-generic-qemu', 'closeMin'): [0, 90000],
+            ('ia32-generic-qemu', 'closeAvg'): [0, 90000],
+            ('ia32-generic-qemu', 'closeMax'): [0, 90000],
+            ('armv7a9-zynq7000-zturn', 'creatMin'): [],
+            ('armv7a9-zynq7000-zturn', 'creatAvg'): [],
+            ('armv7a9-zynq7000-zturn', 'creatMax'): [],
+            ('armv7a9-zynq7000-zturn', 'writeMin'): [],
+            ('armv7a9-zynq7000-zturn', 'writeAvg'): [],
+            ('armv7a9-zynq7000-zturn', 'writeMax'): [],
+            ('armv7a9-zynq7000-zturn', 'closeMin'): [],
+            ('armv7a9-zynq7000-zturn', 'closeAvg'): [],
+            ('armv7a9-zynq7000-zturn', 'closeMax'): []
+        }
+
+    while True:
+        idx = dut.expect([
+            NAME,
+            MSG_LINE,
+            FINAL,
+            END
+        ], timeout=50)
+        parsed = dut.match.groupdict()
+
+        if idx == 0:
+            test_name = parsed["name"]
+            test_name = test_name[test_name.find("bin/") + 4:]
+        elif idx == 1:
+            splited_line = parsed["line"].split()
+            f_use = splited_line[0]
+            count = splited_line[1]
+            size = splited_line[2]
+            files_sec = splited_line[3]
+            app_overhead = splited_line[4]
+
+            dict = {}
+            dict['creatMin'] = splited_line[5]
+            dict['creatAvg'] = splited_line[6]
+            dict['creatMax'] = splited_line[7]
+            dict['writeMin'] = splited_line[8]
+            dict['writeAvg'] = splited_line[9]
+            dict['writeMax'] = splited_line[10]
+            dict['closeMin'] = splited_line[17]
+            dict['closeAvg'] = splited_line[18]
+            dict['closeMax'] = splited_line[19]
+
+            test_status = Status.OK
```

Comment:
I understand that this test is equal to one test case. However, I would move it higher out of the loop (when creating a variable?) or do we even need two variable to show the status in this situation ?